src/Advection/weno5.jl

```diff
@@ -0,0 +1,274 @@
+    if k > 2 && k < grid.Nz - 1
+        return momentum_flux_vw_weno5(i, j, k, grid, v, w)
+    else
+        return momentum_flux_vw(i, j, k, grid, centered_second_order, v, w)
+    end
+end
+
+@inline function momentum_flux_wu(i, j, k, grid::AbstractGrid{FT, <:Bounded}, ::WENO5, u, w) where FT
+    if i > 2 && i < grid.Nx - 1
+        return momentum_flux_wu_weno5(i, j, k, grid, u, w)
+    else
+        return momentum_flux_wu(i, j, k, grid, centered_second_order, u, w)
+    end
+end
+
+@inline function momentum_flux_wv(i, j, k, grid::AbstractGrid{FT, TX, <:Bounded}, ::WENO5, v, w) where {FT, TX}
+    if j > 2 && j < grid.Ny - 1
+        return momentum_flux_wv_weno5(i, j, k, grid, v, w)
+    else
+        return momentum_flux_wv(i, j, k, grid, centered_second_order, v, w)
+    end
+end
+
+@inline function momentum_flux_ww(i, j, k, grid::AbstractGrid{FT, TX, TY, <:Bounded}, ::WENO5, w) where {FT, TX, TY}
+    if k > 2 && i < grid.Nz - 1
+        return momentum_flux_ww_weno5(i, j, k, grid, w)
+    else
+        return momentum_flux_ww(i, j, k, grid, centered_second_order, w)
+    end
+end
+
+#####
+##### Advective tracer fluxes
+#####
+
+@inline advective_tracer_flux_x_weno5(i, j, k, grid, u, c) = Ax_ψᵃᵃᶠ(i, j, k, grid, u) * weno5_flux_x(i, j, k, c)
+@inline advective_tracer_flux_y_weno5(i, j, k, grid, v, c) = Ay_ψᵃᵃᶠ(i, j, k, grid, v) * weno5_flux_y(i, j, k, c)
+@inline advective_tracer_flux_z_weno5(i, j, k, grid, w, c) = Az_ψᵃᵃᵃ(i, j, k, grid, w) * weno5_flux_z(i, j, k, c)
+
+@inline advective_tracer_flux_x(i, j, k, grid, ::WENO5, u, c) = advective_tracer_flux_x_weno5(i, j, k, grid, u, c)
+@inline advective_tracer_flux_y(i, j, k, grid, ::WENO5, v, c) = advective_tracer_flux_y_weno5(i, j, k, grid, v, c)
+@inline advective_tracer_flux_z(i, j, k, grid, ::WENO5, w, c) = advective_tracer_flux_z_weno5(i, j, k, grid, w, c)
+
+@inline function advective_tracer_flux_x(i, j, k, grid::AbstractGrid{FT, <:Bounded}, ::WENO5, u, c) where FT
+    if i > 2 && i < grid.Nx - 1
+        return advective_tracer_flux_x_weno5(i, j, k, grid, u, c)
+    else
+        return advective_tracer_flux_x(i, j, k, grid, centered_second_order, u, c)
+    end
+end
+
+@inline function advective_tracer_flux_y(i, j, k, grid::AbstractGrid{FT, TX, <:Bounded}, ::WENO5, v, c) where {FT, TX}
+    if j > 2 && i < grid.Ny - 1
+        return advective_tracer_flux_y_weno5(i, j, k, grid, v, c)
+    else
+        return advective_tracer_flux_y(i, j, k, grid, centered_second_order, v, c)
+    end
+end
+
+@inline function advective_tracer_flux_z(i, j, k, grid::AbstractGrid{FT, TX, TY, <:Bounded}, ::WENO5, w, c) where {FT, TX, TY}
+    if k > 2 && i < grid.Nz - 1
+        return advective_tracer_flux_z_weno5(i, j, k, grid, w, c)
+    else
+        return advective_tracer_flux_z(i, j, k, grid, centered_second_order, w, c)
+    end
+end
+
+#####
+##### Need to advect momentum like tracers
+#####
+
+@inline div_ũu(i, j, k, grid, advection::WENO5, U) = div_uc(i, j, k, grid, advection, U, U.u)
```

Comment:
As mentioned previously, I can't make sense of this...

Reply:
Has been fixed by pair programming.